packages/api/src/controllers/animalBatchController.js

```diff
@@ -0,0 +1,150 @@
+          .where({ farm_id })
+          .whereNotDeleted()
+          .withGraphFetched('animal_batch_sex_detail');
+        return res.status(200).send(rows);
+      } catch (error) {
+        console.error(error);
+        return res.status(500).json({
+          error,
+        });
+      }
+    };
+  },
+
+  addAnimalBatches() {
+    return async (req, res) => {
+      const trx = await transaction.start(Model.knex());
+
+      try {
+        const { farm_id } = req.headers;
+        const result = [];
+
+        if (!Array.isArray(req.body)) {
+          await trx.rollback();
+          return res.status(400).send('Request body should be an array');
+        }
+
+        for (const animalBatch of req.body) {
+          if (animalBatch.custom_type_id) {
+            const customType = await CustomAnimalTypeModel.query().findById(
+              animalBatch.custom_type_id,
+            );
+            if (customType && customType.farm_id !== farm_id) {
+              await trx.rollback();
+              return res.status(403).send('Forbidden custom type does not belong to this farm');
+            }
+          }
+
+          if (animalBatch.default_breed_id && animalBatch.default_type_id) {
+            const defaultBreed = await DefaultAnimalBreedModel.query().findById(
+              animalBatch.default_breed_id,
+            );
+
+            if (defaultBreed && defaultBreed.default_type_id !== animalBatch.default_type_id) {
+              await trx.rollback();
+              return res.status(400).send('Breed does not match type');
+            }
+          }
+
+          if (animalBatch.custom_breed_id) {
+            const customBreed = await CustomAnimalBreedModel.query()
+              .whereNotDeleted()
+              .findById(animalBatch.custom_breed_id);
+
+            if (customBreed && customBreed.farm_id !== farm_id) {
+              await trx.rollback();
+              return res.status(400).send('Forbidden custom breed does not belong to this farm');
+            }
+
+            if (
+              customBreed.default_type_id &&
+              customBreed.default_type_id !== animalBatch.default_type_id
+            ) {
+              await trx.rollback();
+              return res.status(400).send('Breed does not match type');
+            }
+
+            if (
+              customBreed.custom_type_id &&
+              customBreed.custom_type_id !== animalBatch.custom_type_id
+            ) {
+              await trx.rollback();
+              return res.status(400).send('Breed does not match type');
+            }
+          }
+
+          if (animalBatch.count < 2) {
+            await trx.rollback();
+            return res.status(400).send('Batch count must be greater than 1');
+          }
+
+          if (animalBatch.animal_batch_sex_detail) {
+            let sexCount = 0;
+            const sexIds = [];
+            animalBatch.animal_batch_sex_detail.forEach((sexDetail) => {
```

Comment:
Right, uhm I definitely went with what made sense to me at the time haha and didn't even look at the signature 😂  woops! 

I think I like the practice of matching post body with table column names otherwise yeah I think transforming is wasted effort in my opinion. And I think you can add multiple of the same key see successful request below.. it just overrides the other key: 
![Screenshot 2024-02-02 at 4 07 35 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/41e9f512-5523-480f-93ae-e6857aee50b3)

So I am happy to change that signature but will probably keep it for now unless you definitely want it changed.

Reply:
I'm good with that! Make sure you update the Jira description for verification